src/Parser/RestorationShaman/Modules/Features/AncestralVigor.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import makeWclUrl from 'common/makeWclUrl';
+import SpellIcon from 'common/SpellIcon';
+
+import LazyLoadStatisticBox from 'Main/LazyLoadStatisticBox';
+
+import Module from 'Parser/Core/Module';
+
+const ANCESTRAL_VIGOR_INCREASED_MAX_HEALTH = 0.1;
+const HP_THRESHOLD = 1 - 1/(1 + ANCESTRAL_VIGOR_INCREASED_MAX_HEALTH);
+// this threshold was used to filter the big damages that has more chance to kill a player if there is no ancestral vigor buff.
+const BIG_DAMAGE_THRESHOLD = 0.2;
+
+class AncestralVigor extends Module {
+
+  loaded = false;
+  totalLifeSaved = 0;
+  totalLifeSavedFromBigDamage = 0;
+  on_initialized() {
+    if (!this.owner.error) {
+      this.active = !!this.owner.selectedCombatant.hasTalent(SPELLS.ANCESTRAL_VIGOR_TALENT.id);
+    }
+  }
+
+  // recursively fetch events until no nextPageTimestamp is returned
+  fetchAll(pathname, query) {
+    const self = this;
+    async function checkAndFetch(_query) {
```

Comment:
I've test this branch on my windows PC with IE 11, it works.
I think the CRA project already has babel transform plugin for `async/await`.